nemo/collections/common/tokenizers/text_to_speech/tts_tokenizers.py

```diff
@@ -546,8 +553,14 @@ def __init__(
         if hasattr(g2p, "phoneme_probability"):
             self.phoneme_probability = g2p.phoneme_probability
 
-        # Build tokens list
-        tokens = set(g2p.symbols)
+        # Build tokens list if fixed_vocab isn't set
+        if fixed_vocab:
+            tokens = set(fixed_vocab)
```

Comment:
I wonder if this function takes care of the case that, e.g. 'ö' can be encoded as
    b'\xc3\xb6' (one char) as well as b'o\xcc\x88' (two chars). We discussed similar long time ago. https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/common/tokenizers/text_to_speech/tokenizer_utils.py#L96-L101

Reply:
I'm assuming here that it's reasonable to assume that the user has passed in a "correct"/"canonical" version of the symbols they want (mostly I'm assuming they're copy/pasting from a previous config or model).

Are you suggesting we run normalize over the user input fixed vocab?

Reply:
hmm...i was thinking to apply the same process (calling `normalize_unicode_text`) as what we did now. But this process is applied to `g2p/modules.py` in our current implementation rather than in `tts_tokenizers.py`. I guess `replace_symbols` func should be a better place?

Reply:
@rlangman for better comments.

Reply:
Right now it is done both in `tts_tokenizers.py` as part of `text_preprocessing_func`, as well as in `g2p/modules.py`. I would favor putting any text normalization in `tts_tokenizers.py` where possible.

Reply:
Alright, added a bit of code to preprocess the fixed vocab symbols in the tokenizer init.